pkg/solana/chainwriter/transform_registry_test.go

```diff
@@ -523,39 +557,128 @@ func verifyTxOpts(t *testing.T, options []txmutils.SetTxConfig, exec bool, overh
 
 	if exec {
 		options[1](txConfig)
-		require.Equal(t, overhead+userCU+destGasAmount, txConfig.ComputeUnitLimit)
+		require.Equal(t, overhead+userCU+destGasAmounts, txConfig.ComputeUnitLimit)
 	}
 }
 
-func mockWritableIndexes(t *testing.T, rw *clientmocks.ReaderWriter, tokenAdminRegistryAddr solana.PublicKey) {
-	lookupTablePubkey := GetRandomPubKey(t)
-	tokenAdminRegistry := ccip_common.TokenAdminRegistry{
-		Version:              1,
-		Administrator:        GetRandomPubKey(t),
-		PendingAdministrator: GetRandomPubKey(t),
-		LookupTable:          lookupTablePubkey,
-		// set all accounts as writable
-		WritableIndexes: [2]ag_binary.Uint128{{Endianness: ag_binary.LE, Lo: math.MaxUint64, Hi: math.MaxUint64}},
+func mockExecuteAccountDerivation(t *testing.T, rw *clientmocks.ReaderWriter, offrampStr string, sourceChainSel uint64, userMessagingAccounts []solana.PublicKey, ttAccounts []tokenTransferAccounts, logicReceiver solana.PublicKey, lookupTables []solana.PublicKey) {
+	recentBlockHash := solana.Hash{}
+	rw.On("LatestBlockhash", mock.Anything).Return(&rpc.GetLatestBlockhashResult{Value: &rpc.LatestBlockhashResult{Blockhash: recentBlockHash, LastValidBlockHeight: uint64(100)}}, nil).Once()
+	mockGatherBasicInfoStage(t, rw, offrampStr, sourceChainSel)
+	mockMainAccountListStage(t, rw, offrampStr, userMessagingAccounts, logicReceiver, ttAccounts)
+	mockRetrieveLUTStage(t, rw, offrampStr, ttAccounts)
+	mockTokenTransferStages(t, rw, offrampStr, ttAccounts, lookupTables)
```

Comment:
Other than the Static token transfer stage, which is important for token indices, the rest of the stages here are a bit of an implementation detail onchain, and it would be good not to couple them to this repo, so they don't have to be maintained if the stages change onchain.

It's OK if these are part of the mock mainly for organization purposes, but maybe we can leave a comment saying that it's fine if these drift from their onchain counterparts.

Reply:
Good point, I added the note